### ISSUE TITLE
Getting  x509: certificate signed by unknown authority 

### ISSUE THREAD

**Issue by jainpratik163** [elastic/beats#29144]
I am trying to install metricbeat on my K8s server and  i am getting  **this error  x509: certificate signed by unknown authority**.
and also in values file i mentioned this but still it is checking certificate.

```
ssl:
        verification_mode: "none"
```

i want to know what i am doing wrong or how i can pass my CA certificate 

below is my values file contain

```
daemonset:
  overrideConfig:
    metricbeat.config.modules:
      path: ${path.config}/modules.d/*.yml
      reload.enabled: false
    processors:
      - add_cloud_metadata:
    output.elasticsearch:
      hosts: ['${ELASTICSEARCH_HOST:elasticsearch}:${ELASTICSEARCH_PORT:9201}']
deployment:
  overrideConfig:
    metricbeat.config.modules:
      path: ${path.config}/modules.d/*.yml
      reload.enabled: false
    processors:
      - add_cloud_metadata:
    output.elasticsearch:
      hosts: ['${ELASTICSEARCH_HOST:elasticsearch}:${ELASTICSEARCH_PORT:9201}']
      ssl:
        verification_mode: "none"

```
i am deploying this docker image **metricbeat-oss:6.7.0**

**Comment by jainpratik163** [comment]
> This issue doesn't have a `Team:<team>` label.

Team: Metricbear

**Comment by legoguy1000** [comment]
It looks like you don't have `ssl.verification_mode: none` for the Deamonset, just the deployment.  That could be where its tripping up.

**Comment by jainpratik163** [comment]
Thanks when i add this i am able to connect such configuration we should not do in PROD , so in this case how we can pass CA certs and all.
can you share eg for this.

**Comment by legoguy1000** [comment]
Add ```
ssl.certificate_authorities:
  - <path to CA>
```

**Comment by jainpratik163** [comment]
like this, we need to add
EG:
```
output.elasticsearch.ssl.certificate_authorities: ["<path of ca.pem>"]
output.elasticsearch.ssl.certificate: "<path of cert.pem>"
output.elasticsearch.ssl.key: "/etc/client/cert.key"
```

**Comment by legoguy1000** [comment]
Yes for the certificate authorities setting. The other 2 are only if you're using pki authentication, but if u are, that looks right too.

**Comment by a1exus** [comment]
was this ever resolved? as i'm getting same thing w/ 7.17.X and there I can't find anything in documentation either(
